model/pdata/logs.go

```diff
@@ -96,19 +96,6 @@ func (ld Logs) ResourceLogs() ResourceLogsSlice {
 	return newResourceLogsSlice(&ld.orig.ResourceLogs)
 }
 
```

Comment:
Note for future maintainers: it would be helpful to identify the version in which a Deprecated message was added. This would help us ensure a version has been released that includes the deprecation message without having to dig through the git history. Any thoughts on how this would be best captured?

Reply:
Maybe right in the comment?

```
// Deprecated as of 0.42.0: LogRecordSlice
```

Reply:
@dmitryax please update contributions guide with this. Thanks

Reply:
What I suggested doesn't work because prefix `// Deprecated:` is required. Prepared a PR with a similar syntax that fits the requirement https://github.com/open-telemetry/opentelemetry-collector/pull/4822